pkg-manager/resolve-dependencies/src/resolveDependencies.ts

```diff
@@ -554,6 +554,12 @@ async function resolveDependenciesOfImporterDependency (
       ...importer.options,
       parentPkgAliases: importer.parentPkgAliases,
       pickLowestVersion: pickLowestVersion && !importer.updatePackageManifest,
+      // Cataloged dependencies cannot be upgraded yet since they require
+      // updating the pnpm-workspace.yaml file. This will be handled in a future
+      // version of pnpm.
+      updateToLatest: catalogLookup != null
+        ? false
+        : importer.options.updateToLatest,
```

Comment:
Another approach I was considering was updating the filter here to ignore cataloged dependencies.

https://github.com/pnpm/pnpm/blob/0406d4ad3d2a10326cd46ccc242aadad878a5ead/pkg-manager/plugin-commands-installation/src/recursive.ts#L194-L203

I settled on the current approach since computing the filter list for each project is more involved and not that much less hacky than the workaround above.